Nmap version detection (#1337)
Codecademy/docs: content/cybersecurity/concepts/nmap/terms/nmap-version-detection/nmap-version-detection.md

---
Title: 'Nmap Version Detection'
Description: 'Version detection provides information on all identified open ports which interrogating the ports using probes that are understood by specific services.'
Subjects:
  - 'Computer Science'
  - 'Networking'
Tags:
  - 'Ethical Hacking'
  - 'Cybersecurity'
Catalog/content:
  - 'introduction-to-ethical-hacking'
  - 'paths/scan-systems-with-nmap'
---

**Version detection** provides information on all identified open ports interrogating them using probes that generate specific responses by specific services. Version Detection is done with the `-sV` command and can identify the version number, the service type, the operating system, the hostname, etc.

> The command can be customized with the following options:

Reviewer: ```suggestion

The command can be customized with the following options:
```

>     * --allports

Reviewer 2: ```suggestion
- --allports
```

>     * --version-intensity

Reviewer 2: ```suggestion
- --version-intensity
```

>     * --version-light

Reviewer 2: ```suggestion
- --version-light
```

>     * --version-all

Reviewer 2: ```suggestion
- --version-all
```

>     * --version-trace

Reviewer 2: ```suggestion
- --version-trace
```

Author: ok thanks

## Syntax

```shell

> nmap -sV <localhost>

Reviewer: Apologies, I added them back on since it looks like the style is consistent with the `<` and `>`

```

The command starts an assessment on "localhost".

## Example

> The following example demonstrates the use of Version Detection using `_scanme.nmap.org_`:

Author: Ok, i 'm sorry I didn't realize it before

> The following example demonstrates the use of Version Detection using `_scanme.nmap.org_`:

Reviewer: That's alright, thanks for noticing!

---
Title: 'Nmap Version Detection'

> Description: 'Version detection provides information on all identified open ports which interrogating the ports using probes that are understood by specific services.'

Reviewer: ```suggestion
Description: 'Provides information on all identified open ports which interrogating the ports using probes that are understood by specific services.'
```

---
Title: 'Nmap Version Detection'
Description: 'Version detection provides information on all identified open ports which interrogating the ports using probes that are understood by specific services.'
Subjects:
  - 'Computer Science'
  - 'Networking'
Tags:
  - 'Ethical Hacking'
  - 'Cybersecurity'
Catalog/content:
  - 'introduction-to-ethical-hacking'
  - 'paths/scan-systems-with-nmap'
---

**Version detection** provides information on all identified open ports interrogating them using probes that generate specific responses by specific services.

## Syntax

Version Detection is used with the `-sV` command, and it allows the user to collect information about the port. This can include: the version number, the service type, the operating system, the hostname, etc.

```pseudo
nmap -sV localhost
```

> The command starts an assessment on `localhost`.

Reviewer 2: ```suggestion

The command starts an assessment on `localhost`.
```

---
Title: 'Nmap Version Detection'
Description: 'Version detection provides information on all identified open ports which interrogating the ports using probes that are understood by specific services.'
Subjects:
  - 'Computer Science'
  - 'Networking'
Tags:
  - 'Ethical Hacking'
  - 'Cybersecurity'
Catalog/content:
  - 'introduction-to-ethical-hacking'
  - 'paths/scan-systems-with-nmap'
---

**Version detection** provides information on all identified open ports interrogating them using probes that generate specific responses by specific services.

The command can be customized with the following options:
- `--allports`
- `--version-intensity`
- `--version-light`
- `--version-all`

> - `--version-trace`

Author: ok
